### PR TITLE
Standardize makefiles and add phan, phpstan

### PR DESCRIPTION
**Pull request by skshetry**
- [x]  Add phan
- [x] Add phpstan
- [x] Standardize makefiles

Part of #229

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -85,6 +85,24 @@ pipeline:
       matrix:
         TEST_SUITE: phpunit
 
+  phan:
```
Comment (phil-davis) [inline .drone.yml]: there is a ``php-phan`` section in ``services`` - please investigate, probably delete it

File: .drone.yml [hunk 2]
```diff
@@ -85,6 +85,24 @@ pipeline:
...
+    image: owncloudci/php:${PHP_VERSION}
+    pull: true
+    commands:
+      - make test-php-phan
+    when:
+      matrix:
+        TEST_SUITE: phan
+
+  phpstan:
```
Comment (phil-davis) [inline .drone.yml]: there is a ``phpstan`` section in ``services`` - please investigate, probably delete it

File: .drone.yml [hunk 3]
```diff
@@ -182,18 +200,10 @@ matrix:
   include:
     # phan
     - PHP_VERSION: 7.2
-      TEST_SUITE: phan
-
-    # phpstan
-    - PHP_VERSION: 7.2
-      TEST_SUITE: phpstan
```
Comment (phil-davis) [inline .drone.yml]: This tried to run ``phpstan``. See if it worked, if so then keep it.

File: .drone.yml [hunk 4]
```diff
@@ -182,18 +200,10 @@ matrix:
   include:
     # phan
     - PHP_VERSION: 7.2
-      TEST_SUITE: phan
```
Comment (phil-davis) [inline .drone.yml]: This tried to run ``phan``. See if it worked, if so then keep it.

File: .drone.yml (hunk 3 above)
Comment (skshetry) [reply .drone.yml]: It didn't. I added that myself but had not been configured.

Comment (skshetry) [reply .drone.yml]: This would run this: https://drone.owncloud.com/owncloud/richdocuments/49/21

File: .drone.yml (hunk 2 above)
Comment (skshetry) [reply .drone.yml]: I'll remove that. They never worked

File: .drone.yml (hunk 1 above)
Comment (skshetry) [reply .drone.yml]: Removed.

**Comment by skshetry** [comment]
@phil-davis  done. Please review again

File: .drone.yml [hunk 5]
```diff
@@ -201,85 +201,85 @@ matrix:
...
 
     - PHP_VERSION: 7.2
       OC_VERSION: daily-stable10-qa
       TEST_SUITE: phpunit
       DB_TYPE: mysql
       DB_NAME: owncloud
-      NEED_SERVER: true
-      NEED_INSTALL_APP: true
+      NEED_CORE: true
+      NEED_INSTALL_APP: trueNEED_CORE
```
Comment (phil-davis) [inline .drone.yml]: Random ```NEED_CORE`` pasted here

**Pull request merged**